src/components/checkbox/checkbox.js

```diff
@@ -156,11 +175,18 @@ function MdCheckboxDirective(inputDirective, $mdAria, $mdConstant, $mdTheming, $
           listener(ev);
         }
       }
+
       function listener(ev) {
         if (element[0].hasAttribute('disabled')) {
           return;
         }
 
+        if (indeterminateStateEnabled) {
+          if (attr.indeterminateClick) {
+            scope.$apply(element.attr('indeterminate-click'));
```

Comment:
Also I think, that using `$apply` for that is not the right way.

Why aren't you using `$parse(attr.indeterminateClick)(scope)`?